File: Napakalaki_Java/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author l
 */



public class Monster {


    Prize prize;
    BadConsequence badConsequence;
    private String name;
    private int combatLevel;
    
    Monster(String n, int l, BadConsequence b, Prize p){
        this.name=n;
        this.combatLevel=l;
        this.prize=p;
        this.badConsequence=b;
        
    }
    
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public BadConsequence getBadConsequence(){
        return badConsequence;
    }
    
    public Prize getPrize(){
        return prize;
    }
    public int getLevelsGained(){
        return prize.getLevels();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
    }
    
    public boolean kills(){
        if(badConsequence.myBadConsequenceIsDeath())
            return true;
        else return false;
    }
    
    public String toString(){
        

        return "name= " + name + " Prize= ( " + prize.toString() + " ) combatLevel = " +
                Integer.toString(combatLevel) + " BadConsequence= ( " + badConsequence.toString() + " )";
 
        
    }
    
}

File: Napakalaki_Ruby/monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'Card_dealer.rb'
require_relative 'Bad_consequence.rb'
require_relative 'CombatResult.rb'
require_relative 'dice.rb'
require_relative 'monster.rb'
require_relative 'Napakalaki.rb'
require_relative 'player.rb'
require_relative 'prize.rb'
require_relative 'treasure.rb'
require_relative 'treasure_kind.rb'
module NapakalakiGame
class Monster
  attr_reader :name, :combatLevel, :badConsequence
  def initialize ( n, l,b,  p)
    @name=n
    @combatLevel=l
    @prize=p
    @badConsequence=b
  end
  
  
  def getLevelsGained
    return @prize.levels
  end
  
  def getName
    return @name
  end
  
  def getBadConsequence
    return @badConsequence
  end
  
  def getCombatLevel
    return @combatLevel
  end
  
  def getTreasuresGained
    return @prize.treasures
  end
  
  
  def kills
    return @badConsequence.myBadConsequenceIsDeath
  end
  
  def to_s
    return "name= #@name  combatLevel=#@combatLevel  BadConsequence= ( " +
    @badConsequence.to_s +
     " )  Prize= ( " +
    @prize.to_s  +
     " )"
  end
  
  
end

end